internal/types/compare.go

```diff
@@ -70,7 +123,7 @@ func Compare(docValue, filterValue any) CompareResult {
 				continue
 			}
 
-			if res := compareScalars(docValue, filterValue); res != Incomparable {
+			if res := compareScalars(docValue, filterValue); cond.satisfies(res) {
```

Comment:
I didn't use Max or Min, this somehow got quite a bit complicated. For example comparing $gte with doc bson.A{"c", "b", "a"} with filter bson.D{{"v", bson.D{{"$gte", 41.13}}}}, I think it is greater because [string is greater than number](https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#comparison-sort-order). But mongo DB doesn't think so according to the compat test. Anyway for now, let's skip max and min.